bwe/RateController.cpp

```diff
@@ -298,12 +327,24 @@ RateController::BacklogAnalysis RateController::analyzeBacklog(uint32_t reportNt
             item.delaySinceSR > ntp32Second / 20)
         {
             auto reportCandidate = analyzeProbe(i, modelBandwidthKbps);
-            if (isGood(reportCandidate) || report.lossCount > 0)
+            if (reportCandidate.lossCount > 0 && report.empty())
```

Comment:
Why we stop as soon we have a candidate loss count? If we already have a goodReport when detect another with a lost count (assuming a good report is not not empty), we continue to search. But if we have detect a loss on the 1st candidate, we will abort the search

Reply:
The reasoning is that if we find a loss report it is the most recent condition. In history there may be another report with no loss and if we pick the better one, we will not back off if we detect loss. At least not until the good report has been shifted out.

Reply:
Just to confirm if I understood.

Receive 2 reports:
1. with loss
2. no loss 

1. is the last condition and we should react even if it is not best report according to `bestReport` method).

Receive 2 reports:
1. no loss -> (better report according to `bestReport` method
2. loss

We will not consider the loss report but is not to important because is not the most recent condition (meaning that that lost could be only a small glitch since the 2nd which is a better report and the most recent condition has not reported a lost).

Did I understood the logic correctly?

Reply:
yes, it is problem right if we receive loss, react to that. Then there is another good report. Now that we search for better reports the loss will be picked again. We will not act on it since there is a check later if the report is the same as the one we acted on previously.
Before we just checked the last report, but that one is likely to have a lot of unconfirmed packets. A previous report may on the other hand have a lot more confirmed packets.

Perhaps we need to reconsider the lagorithm.